test/unit/uk/gov/hmrc/exports/services/mapping/AuthorisationHoldersBuilderSpec.scala

```diff
@@ -48,41 +51,130 @@ class AuthorisationHoldersBuilderSpec extends UnitSpec with ExportsDeclarationBu
         )
         val declaration = new Declaration()
 
-        // When
-        new AuthorisationHoldersBuilder().buildThenAdd(model, declaration)
+        authorisationHoldersBuilder.buildThenAdd(model, declaration)
 
-        // Then
         declaration.getAuthorisationHolder must have(size(2))
         declaration.getAuthorisationHolder.get(0).getID.getValue mustBe "eori1"
         declaration.getAuthorisationHolder.get(1).getID.getValue mustBe "eori2"
         declaration.getAuthorisationHolder.get(0).getCategoryCode.getValue mustBe "auth code1"
         declaration.getAuthorisationHolder.get(1).getCategoryCode.getValue mustBe "auth code2"
       }
 
-      "auth code is empty" in {
-        // Given
+      "the ExportsDeclaration instance contains an empty auth code" in {
         val model = aDeclaration(withDeclarationHolders(DeclarationHolder(None, Some("eori"), Some(EoriSource.OtherEori))))
         val declaration = new Declaration()
 
-        // When
-        new AuthorisationHoldersBuilder().buildThenAdd(model, declaration)
+        authorisationHoldersBuilder.buildThenAdd(model, declaration)
 
-        // Then
         declaration.getAuthorisationHolder mustBe empty
       }
 
-      "eori is empty" in {
-        // Given
+      "the ExportsDeclaration instance contains an empty eori" in {
         val model = aDeclaration(withDeclarationHolders(DeclarationHolder(Some("auth code"), None, Some(EoriSource.OtherEori))))
         val declaration = new Declaration()
 
-        // When
-        new AuthorisationHoldersBuilder().buildThenAdd(model, declaration)
+        authorisationHoldersBuilder.buildThenAdd(model, declaration)
 
-        // Then
         declaration.getAuthorisationHolder mustBe empty
       }
     }
   }
 
+  "AuthorisationHolders" when {
+    "the ExportsDeclaration instance is of 'Arrived' type and" when {
+      val additionalType = withAdditionalDeclarationType(STANDARD_FRONTIER)
+
+      "the user didn't enter 'EXRR', 'MIB' or 'CSE' on /add-authorisation-required and" when {
```

Comment:
Is it worth adding a test where the user has added one of these codes and checking that EXRR does not get added to the payload twice?

Reply:
:+1: Good idea